COMPATIBILITY.md

```diff
@@ -0,0 +1,17 @@
+# Compatibility
+
+Volta currently tests against the following platforms, and will treat it as a breaking change to drop support for them:
+
+- macOS
+    - x86-64
+    - Apple Silicon
+- Linux x86-64
+- Windows x86-64
+
+We compile release artifacts compatible with the following, and likewise will treat it as a breaking change to drop support for them:
+
+- macOS v11
+- RHEL and CentOS v6
```

Comment:
One thing #1611 made me think about: since w/v2.0 we will bump our minimum supported version of RHEL/CentOS to v6, is there any reason to bump the minimum supported macOS version? Apple has not been patching macOS 11 for a while now; macOS 12 is the earliest version they are still doing security releases for. (Technically, [they only guarantee support for the current version](https://support.apple.com/guide/deployment/about-software-updates-depc4c80847a/web), but in practice still provide some degree of security patching back a couple versions; it seems like the usual effective policy at this point is "most recent three major releases".)

The flip side is we are not necessarily *gaining* anything by moving.

Reply:
Yeah, I think bumping the macOS one is good as well! In fact, I'm not sure what our _actual_ minimum supported macOS version is, since we build in CI on `macos-latest`. But in any event, AFAIK we haven't gotten any reports of issues where we aren't supporting an old version on macOS.

Reply:
My mild inclination would be to target 12 for the upcoming change, then. I will leave this as is, so it correctly documents what we do *now*, and then we can update it when we do the 2.0 release, and I will also update our build configuration accordingly. 🎉